src/components/CategoriesTabs.tsx

```diff
@@ -0,0 +1,95 @@
+import { Tabs } from '@mui/material'
+import CategoryBtn from './CategoryBtn'
+import { categoriesData } from '~/data/categories'
+import { useReducer } from 'react'
+
+enum ActionType {
+	SET_ACTIVE_CATEGORY = 'SET_ACTIVE_CATEGORY',
+	SET_ACTIVE_SUB_CATEGORY = 'SET_ACTIVE_SUB_CATEGORY',
+}
+
+interface Action {
+	type: 'SET_ACTIVE_CATEGORY' | 'SET_ACTIVE_SUB_CATEGORY'
+	payload: number
+}
+interface State {
+	activeCategory: number
+	activeSubCategory: number | null
+}
+
+const initialState = {
+	activeCategory: 0,
+	activeSubCategory: null,
+}
+
+const reducer = (state: State, action: Action): State => {
```

Comment:
Protip: you can use this package https://github.com/piotrwitek/typesafe-actions
Not-blocking